cloudinit/sources/DataSourceScaleway.py

```diff
@@ -21,20 +23,16 @@
 from cloudinit import net, sources, url_helper, util
 from cloudinit.event import EventScope, EventType
 from cloudinit.net.dhcp import NoDHCPLeaseError
-from cloudinit.net.ephemeral import EphemeralDHCPv4
+from cloudinit.net.ephemeral import EphemeralDHCPv4, EphemeralIPv6Network
 from cloudinit.sources import DataSourceHostname
+from cloudinit.subp import ProcessExecutionError
 
 LOG = logging.getLogger(__name__)
 
-DS_BASE_URL = "http://169.254.42.42"
+DS_BASE_URLS = ["http://169.254.42.42", "http://[fd00:42::42]"]
```

Comment:
what if we made this into a hash with the protocol as key?

Reply:
Why do that?

Most (if not all) of the datasources that use http have some sort of list of endpoints, and address attempt order needs to be preserved - some data sources may have reason to prefer different address class orders for different reasons.

Reply:
the main reason was because later on we parse the addresses to figure out which one is IPv4 and which one is IPv6. We do this twice.

Reply:
This would also complexify the structure as we may eventually add a DNS resolved endpoint that would resolve in both IPv4 and IPv6.